src/lib/mina_base/account_update.ml

```diff
@@ -1485,11 +1446,13 @@ module Body = struct
         (balance_change.magnitude |> Currency.Amount.to_uint64)
     in
     let nonce =
-      match preconditions.account with
-      | Nonce nonce ->
-          Mina_numbers.Account_nonce.of_uint32 nonce
-      | Full _ | Accept ->
-          failwith "Expected a nonce for fee payer account precondition"
+      if Zkapp_precondition.Account.is_nonce preconditions.account then
```

Comment:
This is logic is not equivalent to what it was before (see other comment).

Reply:
I think the difference is that, previously `Nonce` constructor made sure other fields were `Ignore` but now, the precondition could have `Check` for other fields and `Ignore` for the nonce field in which case it is not equivalent. The function should behave more like, `is_nonce_only` (that was implicit before)

Reply:
My bad, as I look at the way `is_nonce` function is implemented, the logic is equivalent. @nholland94 could elaborate on what your expectation is here?